### DISCUSSION TITLE
Add-PnPListItem failing with "property or field 'Id' has not been initialized"

### DISCUSSION THREAD

**Discussion by sravanthiGITHUB** [pnp/powershell#2142] [Q&A]
### Discussed in https://github.com/pnp/powershell/discussions/2140

<div type='discussions-op-text'>

<sup>Originally posted by **sravanthiGITHUB** July 13, 2022</sup>
We have written a PnP-PowerShell script that creates and updates list items. Intermittently, the script fails with an error at the Add-PnPListItem cmdlet: "The property or field 'Id' has not been initialized. It has not been requested or the request has not been executed. It may need to be explicitly requested."
![powershellscripterror](https://user-images.githubusercontent.com/109225932/178748953-a03a0c32-fa86-40cf-841f-b90fa0847671.jpg)
</div>

**Comment by veronicageek** [comment]
@sravanthiGITHUB - Please open an issue by filling the template so we can the maximum information to try help you. Thanks.

**Reply by KoenZomers** [reply]
Indeed, please fill out the detailed template when creating an issue so we can get some context around what you're doing and with which versions. If you're able to reproduce this issue, please have Fiddler running while you reproduce it. This should give us a very good insight into what's going on. Please do NOT attach any Fiddler traces to your posts, but instead share them 1 on 1 privately.

**Reply by sravanthiGITHUB** [reply]
Hi KoenZomers, Please let me know where to find the detailed template.

**Reply by veronicageek** [reply]
@sravanthiGITHUB - When you select "issues" --> "New Issue", click on the `Get Started`  button, and **fill all the required info, with your own troubleshooting, what steps are you taking, was it working on another version**, etc...

![image](https://user-images.githubusercontent.com/25181757/179990101-66019d63-6662-4996-b7d3-5c4f23b14dda.png)
